### PR TITLE
TOOLS-2309 fix: support restoring more than 1k objects per s3 bucket

### PR DESCRIPTION
**Pull request by dwelch-spike**
Also update copyright and python client test dependency.

### CODE REVIEW

File: src/file_proxy_s3.cc
Comment (jdogmcsteezy) [inline src/file_proxy_s3.cc:300]: might not be possible, but maybe save the result of `outcome.GetResult().GetIsTruncated()` since you use it in the loop.  I am not sure if those function calls are idempotent or not

File: src/file_proxy_s3.cc
Comment (jdogmcsteezy) [inline src/file_proxy_s3.cc]: Just a suggestion. Do we need the variable for later?  It seems more inline with Clayton's style to use the func as a condition. Your call 
```suggestion
	if (!ListAllObjects(client, req, res)) {
```

**Review by jdogmcsteezy — APPROVED** [review]
Looks fine to me.  Just some suggestions.  

I think you should edit the commit and summary before merging and be more specific about what you are fixing.  Backups work fine.  Restores do not

File: src/file_proxy_s3.cc
Comment (dwelch-spike) [reply src/file_proxy_s3.cc:300]: The only functions called on `res` are getters that are idempotent.

File: src/file_proxy_s3.cc
Comment (dwelch-spike) [reply src/file_proxy_s3.cc]: Agreed, changed all instances.

**Pull request merged**
